conjure-go-client/httpclient/client.go

```diff
@@ -55,7 +55,8 @@ type clientImpl struct {
 	errorDecoderMiddleware Middleware
 	recoveryMiddleware     Middleware
 
-	uriScorer      internal.RefreshableURIScoringMiddleware
+	uriPool        internal.URIPool
```

Comment:
`URIPool` is new functionality. I think the behavior can be captured behind a `URISelector` implementation so prefer to just use the refreshable slice of URIs directly for now.